packages/@expo/metro-config/src/serializer/treeShakeSerializerPlugin.ts

```diff
@@ -797,7 +796,11 @@ export async function treeShakeSerializer(
       },
       Identifier(path) {
         // Make sure this identifier isn't coming from an import specifier
-        if (!path.scope.bindingIdentifierEquals(path.node.name, path.node)) {
+        if (
+          !path.scope.bindingIdentifierEquals(path.node.name, path.node) &&
+          // `import { Foo as Bar } from 'bax'` Foo should not be marked as used.
+          path.parent.type !== 'ImportSpecifier'
```

Comment:
Checking only `ImportSpecifier` should be sufficient because `ImportDefaultSpecifier` and `ImportNamespaceSpecifier` create bindings, so the bindingIdentifierEquals => false. 

And case of `ImportSpecifier` -> `import { Foo as Bar } from 'bax'`, Foo is non binding Identifier (and also not a referenced identifier) => should not counted as used identifier.